app/models/host/discovered.rb

```diff
@@ -3,7 +3,7 @@
 class Host::Discovered < ::Host::Base
   include ScopedSearchExtensions
 
-  belongs_to :subnet
+  has_one :subnet, :through => :primary_interface
   belongs_to :hostgroup
```

Comment:
btw: do we care about other subnets? (assuming the host has multiple interfaces)